File: majorProject/src/test/java/com/aocc/majorproject/GameSettingsTest.java
package com.aocc.majorproject;

import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class GameSettingsTest {

    @Test
    public void toggleSound_flipsSoundFlag() {
        GamePreferences.sound = true;

        GameSettings.toggleSound();
        assertFalse(GamePreferences.sound);
        assertTrue(GamePreferences.getTapVolume() == 0);

        GameSettings.toggleSound();
        assertTrue(GamePreferences.sound);
        assertTrue(GamePreferences.getTapVolume() == 10);
    }
}

File: majorProject/src/test/kotlin/com/aocc/majorproject/GameSettingsTest.kt
package com.aocc.majorproject

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class GameSettingsTest {

    @Test
    fun toggleSound_flipsSoundFlag() {
        GamePreferences.sound = true

        GameSettings.toggleSound()
        assertFalse(GamePreferences.sound)
        assertTrue(GamePreferences.getTapVolume() == 0)

        GameSettings.toggleSound()
        assertTrue(GamePreferences.sound)
        assertTrue(GamePreferences.getTapVolume() == 10)
    }
}
